crates/trigger-http/src/lib.rs

```diff
@@ -244,11 +248,20 @@ impl HttpTrigger {
 
         // Handle well-known spin paths
         if let Some(well_known) = path.strip_prefix(spin_http::WELL_KNOWN_PREFIX) {
-            return match well_known {
-                "health" => Ok(Response::new(body::full(Bytes::from_static(b"OK")))),
-                "info" => self.app_info(),
-                _ => Self::not_found(NotFoundRouteKind::WellKnown),
+            let well_known_matched = match well_known {
+                "health" => Some(Ok(Response::new(body::full(Bytes::from_static(b"OK"))))),
+                "info" => Some(self.app_info()),
+                _ => None,
             };
+            if let Some(res) = well_known_matched {
+                current_span.record("otel.name", format!("{} {}", req.method(), path));
+                current_span.record("http.route", path);
+                current_span.record("http.response.status_code", 200);
+                return res;
+            }
+            current_span.record("otel.name", format!("{}", req.method()));
+            current_span.record("http.response.status_code", 404);
```

Comment:
I can no longer follow the flow of this stanza at all.  It's gone from a simple pattern match to big pile of tracing statements whose shape gives me no sense of the underlying logic.  (I know this is not on you and that you are trying to get it done with the tools OpenTelemetry gives you and are probably as queasy as I am!)

Is it possible that moving the tracing up a level could capture this stuff in a unified way?  The various flows here might make it difficult I realise.